### ISSUE TITLE
linter: `unicorn/consistent-function-scoping` misses functions that only reference an ancestor scope

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26591]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
    "plugins": ["unicorn"],
    "rules": {
        "unicorn/consistent-function-scoping": [
            "error",
            { "checkArrowFunctions": false }
        ]
    }
}
```

## Reproduction

```ts
const shared = 1;

function outer() {
    function inner() {
        return shared;
    }

    return inner();
}
```

Run:

```sh
oxlint -c .oxlintrc.json example.ts
```

## Actual behavior

Oxlint reports no violation.

## Expected behavior

Oxlint should report `inner`. It does not capture anything from its immediate parent function and
can be moved one level out while retaining access to `shared`:

```ts
const shared = 1;

function inner() {
    return shared;
}

function outer() {
    return inner();
}
```

`eslint-plugin-unicorn@74.0.0` reports:

```text
Move function 'inner' to the outer scope  unicorn/consistent-function-scoping
```

**Comment by lukpsaxo** [comment]
This more specific case was previously fixed: [#22381](https://github.com/oxc-project/oxc/issues/22381)
